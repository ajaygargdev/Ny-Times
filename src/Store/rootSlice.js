import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { getArticles } from "../Services";

export const fetchArticles = createAsyncThunk(
  "root/fetchArticles",
  async (page = 1) => {
    console.log("-----test----");
    const response = await getArticles(page);
    if (!response.success) {
      throw new Error("Network response was not ok");
    }
    return { ...response.data, page };
  },
);

const initialState = {
  articles: [],
  copyright: "",
  isLoading: false,
  error: null,
  currentPage: "1",
};

export const rootSlice = createSlice({
  name: "root",
  initialState,
  reducers: {
    changePage: (state, payload) => {
      state.currentPage = payload + "";
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchArticles.pending, (state) => {
        state.isLoading = true;
        state.error = null;
      })
      .addCase(fetchArticles.fulfilled, (state, action) => {
        state.isLoading = false;
        state.articles = {
          ...state.articles,
          [action.payload.page]: action.payload?.results || [],
        };
        state.copyright = action.payload.copyright;
      })
      .addCase(fetchArticles.rejected, (state, action) => {
        state.isLoading = false;
        state.error = action.error;
      });
  },
});

export const { changePage } = rootSlice.actions;

export default rootSlice.reducer;
