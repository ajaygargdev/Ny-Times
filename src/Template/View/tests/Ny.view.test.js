/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, screen } from "@testing-library/react";
import NyView from "../Ny.view";
import { store } from "../../../Store/Store";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { Provider } from "react-redux";
import { mockData } from "./Ny.view.mock";

const storeWrapper = (Comp) => {
  return (
    <Provider store={store}>
      <Comp />
    </Provider>
  );
};

describe("Container Component", () => {
  let mockAxios;

  beforeAll(() => {
    mockAxios = new MockAdapter(axios);
  });

  afterAll(() => {
    mockAxios.restore();
  });

  test("renders without crashing", async () => {
    const { container } = render(storeWrapper(NyView));
    expect(container).toMatchSnapshot();
  });

  test("renders with loading state", async () => {
    render(storeWrapper(NyView));
    const loader = screen.getByText(/Loading/i);
    expect(loader).not.toBeNull();
  });

  //   test("renders with footer", async () => {
  //     const responseData = { status: 200, data: mockData };
  //     mockAxios.onGet().reply(200, responseData);
  //     await render(storeWrapper(NyView));
  //     await new Promise((resolve) => setTimeout(resolve, 500));
  //     const footer = await screen.getByText(/copyright/i);
  //     console.log(footer);
  //     expect(footer).not.toBeNull();
  //   });
});
