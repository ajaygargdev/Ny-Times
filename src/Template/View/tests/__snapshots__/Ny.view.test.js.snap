// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders without crashing 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-5iifl-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          The New York Times
        </h6>
      </div>
    </header>
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg css-1m3eoub-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-1lb6n3v"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-ib86xl-MuiTypography-root"
        >
          Loading articles...
        </h6>
      </div>
    </main>
    <footer
      class="MuiBox-root css-y52xi2"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-19zpnxz-MuiTypography-root"
          style="--Typography-textAlign: center;"
        />
      </div>
    </footer>
  </div>
</div>
`;
